### PR TITLE
Marking internal action as @ignore to remove them from the documentation.

### PR DESCRIPTION
**Pull request by ryanwelcher**
## What?
This PR marks internal actions that are not meant to be accessed outside of this store as `@ignore` to remove them from the related docs. 

## Why?
Having these items in the docs just adds noise to the documentation and has no real value. There is a larger conversation around how these are marked but backed on the [JSDoc tags in the documentation standards](https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/#supported-jsdoc-tags) this should be the correct tag to use.

### CODE REVIEW

**Comment by gziolo** [comment]
Should we follow with WordPress core and use `@internal`? The alternative would be to make them with `@deprecated` if we belive they are no longer necessary.

File: packages/core-data/README.md [hunk 1]
```diff
@@ -90,58 +90,6 @@ _Returns_
 
 -   `Object`: Action object.
 
-### receiveAutosaves
```
Comment (gziolo) [inline packages/core-data/README.md:93]: It looks like we print the same content in `@wordpress/core-data`'s README file. Should we link to the reference gude instead?

**Comment by ryanwelcher** [comment]
Now that https://github.com/WordPress/gutenberg/pull/42198 has been merged, we can use `@ignore` as per the [JavaScript Coding standards](https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/#supported-jsdoc-tags)

**Comment by ryanwelcher** [comment]
@youknowriad @adamziel @Mamaduka @mtias I'd love any input you may have on this.

**Comment by Mamaduka** [comment]
The change makes sense to me. The internal actions aren't beneficial if you're not working on the core.

**Comment by gziolo** [comment]
I like the direction. We should stop promoting code that can be expressed in a more general way today. It would be great to leave a not in JSDoc explaining why it's ignored, so it's clear that usage of those functions is discouraged.

**Comment by Mamaduka** [comment]
> It would be great to leave a not in JSDoc explaining why it's ignored, so it's clear that usage of those functions is discouraged.

That's a great idea. Maybe we can do something like - `@ignore [reason]`?

**Comment by ryanwelcher** [comment]
> That's a great idea. Maybe we can do something like - `@ignore [reason]`?

Looks like the linter won't allow this, I get the following when trying to commit.

`error  @ignore should be empty  jsdoc/empty-tags`.

Seems we'd need to make a change to https://github.com/WordPress/gutenberg/blob/trunk/packages/eslint-plugin/configs/jsdoc.js#L128 to allow that. Personally, I really this the idea of having the reason in line with the tag so I'll add that change to this PR.

**Comment by ryanwelcher** [comment]
Based on a conversation in Slack with @gziolo , it looks like core always has the `@ignore` tag empty. I think it makes sense to match that approach here and have added the reason for ignoring to the main description.

**Review by gziolo — APPROVED** [review]
💯

**Pull request merged**